lib/spack/spack/environment/environment.py

```diff
@@ -1974,10 +1983,11 @@ def _read_lockfile_dict(self, d):
 
         # Track specs by their DAG hash, allows handling DAG hash collisions
         first_seen = {}
-
+        current_lockfile_format = d["_meta"]["lockfile-version"]
+        reader = READER_CLS[current_lockfile_format]
```

Comment:
Should this give a nice error? "This version of Spack does not support environments version ..."

Reply:
Done in 0d752d8